packages/mobile/src/transactions/TransactionReview.tsx

```diff
@@ -46,7 +84,9 @@ class TransactionReview extends React.PureComponent<Props> {
   }
 
   getConfirmationProps = () => {
-    const confirmationProps = this.props.route.params.confirmationProps
+    const confirmationProps =
+      this.props.route.params.transferConfirmationProps ||
```

Comment:
Similar question here - how do we know that one and only one of the right type is passed?

Reply:
Admittedly less apparent on this screen but the only way to navigate here is by using either `navigateToPaymentTransferReview` or `navigateToExchangeReview`. Each of these functions requires their respective `confirmationProps` as arguments so it would be impossible to get to this screen with anything other than exactly 1

Reply:
I guess probably convention would be using those functions, but I think in the future someone could add another type to TransactionReview and just call an explicit navigate. Basically, there is nothing stopping someone from just calling `this.props.navigation.navigate(Screens.TransactionReview)`

Reply:
That's true, only did it this way because I thought I had too. Using typeguards is definitely preferable